frontend/src/components/UserTables/RidersContext.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { Rider } from '../../types';
+import { useReq } from '../../context/req';
+
+type ridersState = {
+    getExistingRiders: () => Array<Rider>, 
+    refreshRiders: () => Promise<Array<Rider>> 
+};
+
+const initialState: ridersState = {
+   getExistingRiders: () => [],
+   refreshRiders: async () =>  []
+  };
+const RidersContext = React.createContext(initialState);
+
+export const useRiders = () => React.useContext(RidersContext);
+
+type RidesProviderProps = {
+  children: React.ReactNode;
+}
+
+type RidersType = Array<Rider>;
+
+export const RidesProvider = ({ children }: RidesProviderProps) => {
+  const [riders, setRiders] = useState<RidersType>([])
+  const getExistingRiders = () => {return riders};
+  const { withDefaults } = useReq();
+  const refreshRiders = async () => {
+    const ridersData:RidersType = await fetch('/api/riders', withDefaults())
+        .then((res) => res.json())
+        .then((data) => data.data);
+        
+      setRiders((prev) => ({ ...prev, ridersData})); 
```

Comment:
should be creating a new array here instead of an object, also should be `[ ...ridersData ]` or else we'll end up with duplicate data, so no need for the callback

Reply:
I think it was always creating a new array since RidersType was just an array but that was definitely not needed at all.